src/types/builtin/primitives.rs

```diff
@@ -16,7 +16,7 @@ use super::builtin_type_value::BuiltinTypeValue;
 
 // ----------- A simple u32 type --------------------------------------------
 
-#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+#[derive(Debug, Eq, PartialEq, Copy, Clone, Hash)]
```

Comment:
Why do you derive Hash here for U32 but for U8 do you don't derive it but instead provide a todo!() implementation of the Hash trait?

Reply:
Ah, ignore me, I think I get it.

Reply:
Because I for U8 I have manually implemented PartialEq, and then you can't derive Hash anymore. I should also have a manual PartialEq for U32 also, so then the derive for Hash will also have to go on U32

Reply:
PartialEq is manual btw, because type conversion is done in there